crates/executor/src/lib.rs

```diff
@@ -57,6 +57,7 @@ where
     pub fn new(
         config: &'a RollupConfig,
         parent_header: Sealed<Header>,
+        // TODO: @Clabby - would you rather refactor this to assume Provider implements F + H and just pass one argument?
```

Comment:
Question for you @clabby: We can keep this as fetcher F and hinter H and just use the provider for both, or we can refactor the executor to expect the provider to handle everything. Which would you prefer?

Same question with the TrieDB, which also expects separate fetcher and hinter.